tests/symbols.c

```diff
@@ -0,0 +1,144 @@
+#include <stdio.h>
+#include "zstd_errors.h"
+#define ZSTD_STATIC_LINKING_ONLY
+#include "zstd.h"
+#define ZBUFF_DISABLE_DEPRECATE_WARNINGS
+#define ZBUFF_STATIC_LINKING_ONLY
+#include "zbuff.h"
+#define ZDICT_STATIC_LINKING_ONLY
+#include "zdict.h"
+
+static const void *symbols[] = {
```

Comment:
This is a great way to test presence of symbols.

Obviously, it features the maintenance shortcoming that it needs to be manually edited whenever a new prototype is added.
As it stands, it's very easy to create a new prototype and forget to add it to `symbols.c`, and then test passes correctly...

Unfortunately, I also have no really better alternative to propose.

I was vaguely considering to create a build mode for `tests` programs, that would use dynamic `libzstd` instead of compiling and linking directly `lib` source code.
Of course, there's a bit of the same problem, a new prototype needs to be invoked from at least one of these test programs to check its presence. But it's supposed to be the case, any prototype should have at least one basic unit test associated.